scormxblock/config.py

```diff
@@ -0,0 +1,137 @@
+# -*- coding: utf-8 -*-
+import pkg_resources
+
+
+class _ScromComponent(object):
+    """Hold information of an scrom web content component on Studio page.
+    """
+    TAG_VIDEO = r'Video'
+    TAG_QUIZ = r'Quiz'
+    TAG_ELEARNING_AUTHORING_RECORDER= r'eLearning authoring tools'
+    # The order of items in `ALL_TAGS` would be used as Tabs order in Author_View
+    ALL_TAGS = [TAG_QUIZ, TAG_ELEARNING_AUTHORING_RECORDER, TAG_VIDEO]
+
+    def __init__(self, icon, name, description, tags, paying, site_link, get_scrom_handler):
+        """Constructor of Scrom  Web Content Configuration ( Support Image/Icon `SVG` only )
+            @param icon:    path of Image SVG
+            @type icon:     string
+        """
+        self.icon = icon
+        self.name = name
+        self.description = description
+        self.tags = list(tags) if isinstance(tags, (list, tuple)) else [tags]
+        self.paying = paying
+        self.site_link = site_link
+        self._get_scrom_handler = get_scrom_handler
+
+        if not all([tag in _ScromComponent.ALL_TAGS for tag in self.tags]):
+            raise NameError('Unsupported tags : {}'.format(str(self.tags)))
+    def get_tags_set(self):
+        return set(self.tags)
+
+    @property
+    def svg_image(self):
+        """Return svg image description"""
+        return self._get_scrom_handler().resource_string(self.icon)
+
+    def __str__(self):
+        return self.name
+    
+
+
+class SupportedScromResources(object):
+    """An iterable object definition for listed `Tags` & `Sites`
+    """
+    _scrom_xblock_singleton = None
+
+    def __init__(self):
+        """Initialize scrom resources vector"""
+        self._listed_tags = set()
+        self._resources = []
+
+        self._add_resource(
+            icon='static/images/adope-captivate.svg', name='Adobe Captivate',
+            tags=_ScromComponent.TAG_ELEARNING_AUTHORING_RECORDER,
+            paying=True,
+            site_link=r'https://www.adobe.com/products/captivate.html',
+            description=r'Create stunning courses in minutes.'
+        )
+
+        self._add_resource(
+            icon='static/images/articulate-360.svg', name='Articulate 360',
+            tags=_ScromComponent.TAG_ELEARNING_AUTHORING_RECORDER,
+            paying=True,
+            site_link=r'https://articulate.com/360',
+            description=r'Use Storyline 360 to create courses with custom interactivity or Rise to create responsive courses right in your web browser.'
```

Comment:
? Use Storyline 360 to create courses with custom interactivity or Rise 👉  `360` to create responsive courses right in your web browser.

Reply:
https://github.com/Learningtribes/edx_xblock_scorm/pull/29/commits/ecb1cb34c973d1be44244c64338986238a84a41c

Reply:
👍